src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -19539,23 +19539,18 @@ public void Main(string? key)
 
         [Fact]
         [WorkItem(29858, "https://github.com/dotnet/roslyn/issues/29858")]
-        public void GenericMethod_WithEnsuresNotNullOnReturnType()
+        public void GenericMethod_WithNotNullOnReturnType()
         {
             CSharpCompilation c = CreateCompilation(@"
 using System.Runtime.CompilerServices;
 public class C
 {
-    [EnsuresNotNull]
+    [NotNull]
     public static T Copy<T>(T key) => throw null!;
 }
-" + EnsuresNotNullAttributeDefinition);
+" + NotNullAttributeDefinition);
 
-            c.VerifyDiagnostics(
-                // (5,6): error CS0592: Attribute 'EnsuresNotNull' is not valid on this declaration type. It is only valid on 'parameter' declarations.
```

Comment:
>// (5,6): error CS0592: Attribute 'EnsuresNotNull' is not valid on this declaration type. It is only valid on 'parameter' declarations. [](start = 16, length = 135)

It is not clear why this error is no longer reported.

Reply:
`[NotNull]` is valid on other targets than parameters.

---
In reply to: [286267022](https://github.com/dotnet/roslyn/pull/35846#discussion_r286267022) [](ancestors = 286267022)

Reply:
> [NotNull] is valid on more targets just than parameters.

What else can it be applied to other than parameters and a return value?

---
In reply to: [286307930](https://github.com/dotnet/roslyn/pull/35846#discussion_r286307930) [](ancestors = 286307930,286267022)

Reply:
Fields, properties, ...

---
In reply to: [286597495](https://github.com/dotnet/roslyn/pull/35846#discussion_r286597495) [](ancestors = 286597495,286307930,286267022)